pkg/credentials/filecache_v2.go

```diff
@@ -0,0 +1,110 @@
+package credentials
```

Comment:
@Skarlso has the most context on this, going to defer to him but the testing looks good (athough -1 point for using tables :wink: )

Reply:
Kubernetes and many other popular projects use table-driven testing (with or without Ginkgo), and it's in fact uncommon for those projects to copy-paste code in tests :slightly_smiling_face:

Reply:
> Kubernetes and many other popular projects use table-driven testing

and many projects don't write unit tests at all, doesn't mean its correct :grin: